mlflow/models/evaluation/base.py

```diff
@@ -793,52 +793,82 @@ def evaluate(
+                           - Numpy array. This will be saved as a .npy artifact.
+                           - Matplotlib Figure. This will be saved as an image artifact. Note that
+                             ``matplotlib.pyplot.savefig`` is called behind the scene with default
+                             configurations. To customize, either save the figure with the desired
+                             configurations and return its file path or define customizations
+                             through environment variables in ``matplotlib.rcParams``.
+                           - Any JSON serializable object. This will be saved as a JSON artifact.
 
                            .. code-block:: python
                                :caption: Custom Metric Function Boilerplate
 
                                def custom_metrics_boilerplate(eval_df, builtin_metrics):
                                    # ...
                                    metrics: Dict[AnyStr, Union[int, float, np.number]] = some_dict
+                                   artifacts: Dict[AnyStr, Any] = some_artifact_dict
                                    # ...
+                                   if artifacts is not None:
+                                       return metrics, artifacts
                                    return metrics
 
                            .. code-block:: python
                                :caption: Example usage of custom metrics
 
                                def squared_diff_plus_one(eval_df, builtin_metrics):
-                                 return {
-                                     "squared_diff_plus_one": (
-                                         np.sum(
-                                             np.abs(
-                                                 eval_df["prediction"] - eval_df["target"] + 1
-                                             ) ** 2
-                                         )
-                                     )
-                                 }
+                                   return {
+                                       "squared_diff_plus_one": (
+                                           np.sum(
+                                               np.abs(
+                                                   eval_df["prediction"] - eval_df["target"] + 1
+                                               ) ** 2
+                                           )
+                                       )
+                                   }
+
+                               def scatter_plot(eval_df, builtin_metrics):
+                                   plt.scatter(eval_df['prediction'], eval_df['target'])
+                                   plt.xlabel('Targets')
+                                   plt.ylabel('Predictions')
+                                   plt.title("Targets vs. Predictions")
+                                   plt.savefig("some/path/example.png")
```

Comment:
We need to make sure user can actually write to a path in custom function. Update it with a tempdir + "/example.png" or something functional as is.